base/multidimensional.jl

```diff
@@ -395,11 +422,8 @@ module IteratorsMD
     @inline to_indices(A, inds, I::Tuple{CartesianIndices{0},Vararg{Any}}) =
         (first(I), to_indices(A, inds, tail(I))...)
 
-    @inline function in(i::CartesianIndex{N}, r::CartesianIndices{N}) where {N}
-        _in(true, i.I, first(r).I, last(r).I)
-    end
-    _in(b, ::Tuple{}, ::Tuple{}, ::Tuple{}) = b
-    @inline _in(b, i, start, stop) = _in(b & (start[1] <= i[1] <= stop[1]), tail(i), tail(start), tail(stop))
+    @inline in(i::CartesianIndex, r::CartesianIndices) = false
+    @inline in(i::CartesianIndex{N}, r::CartesianIndices{N}) where {N} = all(map(in, i.I, r.indices))
 
     simd_outer_range(iter::CartesianIndices{0}) = iter
     function simd_outer_range(iter::CartesianIndices)
```

Comment:
I don't know if it's needed to modify this simd part:

https://github.com/JuliaLang/julia/blob/8a9666ae2278c61e021ed21864feb77d04a53dc7/base/multidimensional.jl#L412-L415

Reply:
Probably we're missing a test. "Inner" iteration in a `@simd` goes from `0:n-1`, you'll need to compute the correct index from that.

Reply:
Could you give me some test examples that I can work on? I don't quite know what should be the correct behavior...

Reply:
Changed in 7e1b9aa but I'm not sure if I get things done correctly.

Reply:
6efb112 fixes the simd performance by propogating inbounds check.